### ISSUE TITLE
[FEA] Implement `first` and `last` groupby aggregations

### ISSUE THREAD

**Issue by sarahyurick** [rapidsai/cudf#8966]
In [Pandas](https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L1870) and [Dask](https://github.com/dask/dask/blob/main/dask/dataframe/groupby.py#L720), `first` and `last` are `groupby().agg()` methods, e.g.: 
```
import pandas as pd

df = pd.DataFrame()
df['id'] = [1, 0, 1]
df['val'] = [0, 4, 2]

df.groupby('id').agg("first")
	val
id	
0	4
1	0
```
and in Dask: 
```
import pandas as pd
import dask

df = pd.DataFrame()
df['id'] = [1, 0, 1]
df['val'] = [0, 4, 2]

ddf = dask.dataframe.from_pandas(df, npartitions=1)
ddf.groupby('id').agg(["first"]).compute()
```

I'd like to be able to do the same with cuDF and Dask cuDF. After some initial exploring, I found that `cudf._lib.aggregation.Aggregation` will need to implement the `first` and `last` methods. 

Similar to #1476. 

Finally, note that Pandas defines `first` to grab the `first item that isn't NA` and `last` to grab the `last item that isn't NA`, so `first` for example isn't necessarily the same as `nth(0)`.

**Comment by shwina** [comment]
Note that `first()` (or `last()`) are different from `nth()`, in that they return the first (or last) _non-null_ element in a group:

```python
In [3]: df
Out[3]:
   a    b
0  1  NaN
1  1  2.0
2  1  3.0
3  2  NaN
4  3  5.0

In [4]: df.groupby("a").first()
Out[4]:
     b
a
1  2.0
2  NaN
3  5.0

In [5]: df.groupby("a").nth(0)
Out[5]:
     b
a
1  NaN
2  NaN
3  5.0
```

**Comment by charlesbluca** [comment]
Resolved with #9004
